File: javascript/magnitude/magnitude.js
LOG_BASE1K_DIVSOR = Math.log(1000)

MAGNITUDE_SUFFIXES = ['', 'K', 'M', 'B', 'T']

Magnitude = {
  format: function (number) {
    if (number < 1000) return '' + number;

    var logBase1K = Math.log(number) / LOG_BASE1K_DIVSOR
    var suffixIndex = Math.floor(logBase1K)
    var suffix = MAGNITUDE_SUFFIXES[suffixIndex]
    var shortNum = Math.round(number / Math.pow(1000, suffixIndex))

    return '' + shortNum + suffix
  }
}

File: scala/magnitude/Magnitude.scala
package magnitude

object Magnitude {
  val SUFFIXES = List[String]("", "K", "M", "B")

  def format(num: Long, suffixes: List[String] = SUFFIXES): String =
    suffixes match {
      case suffix::suffixTail =>
        if (num < 1000) {
          num + suffix
        } else {
          format(num / 1000, suffixTail)
        }
      case Nil => num + "T"
    }
}
